pkg/trace/api/profiles.go

```diff
@@ -34,6 +34,13 @@ func profilingEndpoints(apiKey string) (urls []*url.URL, apiKeys []string, err e
 	main := profilingURLDefault
 	if v := config.Datadog.GetString("apm_config.profiling_dd_url"); v != "" {
 		main = v
+		if strings.HasSuffix(main, "/v1/input") {
```

Comment:
I don’t think this is a good approach. _dd_url settings are always supposed to be the full URL. We should not alter it, nor tack things onto it as extra suffix. 

However, you could warn the user that a new API version is available and that he should upgrade in case he is using v1. But this is a very low probability case because this value was already the default and I don’t think any users actually change the dd_url

Reply:
I tend to agree with this. Changing the default endpoint/URL is a no-brainer. Rewriting URLs given by the user seems precarious. Printing a warning seems like a better approach.

Reply:
agreed

Reply:
Potentially a bit late to this party (was on PTO) but

> _dd_url settings are always supposed to be the full URL (..)  I don't think any users actually change the dd_url

This is actually a pretty common setup for clients using a reverse proxy and is part of our haproxy setup documentation: https://docs.datadoghq.com/agent/proxy/?tab=agentv6v7#datadog-agent-configuration

Right now profiling seems to be a bit of an odd-duck in this space as for things to work correctly users do need to specify the `/v1/input` path and they don't need to do this with other intakes (e.g. logs and APM).

I still agree automatically rewriting is not good but maybe adding the path segment if missing would actually align it with other dd_url configs (see https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/writer/sender.go#L41 and https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/writer/trace.go#L107)?